core/src/main/scala/org/bykn/bosatsu/Externals.scala

```diff
@@ -39,6 +39,14 @@ object FfiCall {
     def call(t: rankn.Type): Value = evalFn
   }
 
+  final case class ExprFn(wrapper: (LetFreeEvaluation.LetFreeValue, rankn.Type, LetFreeEvaluation.Cache, LetFreeEvaluation.ToLFV) => Any) extends FfiCall {
+    import LetFreeEvaluation.ExprFnValue
+
+    private[this] def evalExprFn(t: rankn.Type): ExprFnValue = ExprFnValue({ (e1, cache, eval) => Value.ExternalValue(wrapper(e1, t, cache, eval)) })
+
+    def call(t: rankn.Type): Value = new Value.FnValue(evalExprFn(t))
```

Comment:
I'm going to merge this, but I'd like to refactor to remove Value depending on LetFree (which I think is a circular dependency).

For instance, I think you just want a `case class FromFn(callFn: rankn.Type => Value) extends FfiCall { def call(t: rankn.Type) = callFn(t) }`

Reply:
That is much simpler than what I had.